### PR TITLE
Update Dockerfile and README

### PR DESCRIPTION
**Pull request by martialblog**
Updated Dockerfile:

- Add full-qualified name to Base Image, works better with other Container Runtimes
- Update to Focal 20.04
- Use arguments to be more flexible with downloads
- Add Checksum to foreign code download. Hashtag security :safety_vest: 

Added Makefile:

- Simplify Image build
  
Updated README
     
- Remove $ prefix to make commands easier to copy

### CODE REVIEW

**Comment by martialblog** [comment]
Tested with `gitlab-training`

**Comment by martialblog** [comment]
@mocdaniel FYI

**Review by mocdaniel — APPROVED** [review]
Looks good to me, due to the range of changes to the Dockerfile we should get a second review by one of the initial authors, too.

Edit: They're already requested as reviewers.

**Review by mocdaniel — APPROVED** [review]

**Pull request merged**
